hop/auth.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+__author__ = "Patrick Godwin (patrick.godwin@psu.edu)"
+__description__ = "a module for auth utilities"
+
+import argparse
+import errno
+import logging
+import os
+
+from adc import auth
+
+DEFAULT_AUTH_CONFIG = """\
+security.protocol=SASL_SSL
+sasl.mechanism=PLAIN
+sasl.username=username
+sasl.password=password
+"""
+
+logger = logging.getLogger("hop")
+
+# expose auth options from adc
+SASLAuth = auth.SASLAuth
+SASLMethod = auth.SASLMethod
+
+# map default auth option to Auth
+Auth = auth.SASLAuth
+
+
+def get_auth_path():
+    """Determines the default location for auth configuration.
+
+    Returns:
+        The path to the authentication configuration file.
+
+    """
+    auth_filepath = os.path.join("hop", "auth.conf")
+    if "XDG_CONFIG_HOME" in os.environ:
+        return os.path.join(os.getenv("XDG_CONFIG_HOME"), auth_filepath)
+    else:
+        return os.path.join(os.getenv("HOME"), ".config", auth_filepath)
+
+
+def load_auth(authfile=get_auth_path()):
+    """Configures an Auth instance given a configuration file.
+
+    Args:
+        authfile: Path to a configuration file, loading from
+            the default location if not given.
+
+    Returns:
+        A configured Auth instance.
+
+    Raises:
+        KeyError: An error occurred parsing the configuration file.
+
+    """
+    if not os.path.exists(authfile):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), authfile)
+
+    # load config
+    with open(authfile, "r") as f:
+        config = {opt[0]: opt[1] for opt in (line.split("=") for line in f.read().splitlines())}
+
+    # translate config options
+    ssl = True
+    extra_kwargs = {}
+    try:
+        # SSL options
+        if config["security.protocol"] == "SASL_PLAINTEXT":
+            ssl = False
+        elif "ssl.ca.location" in config:
+            extra_kwargs["ssl_ca_location"] = config["ssl.ca.location"]
+
+        # SASL options
+        user = config["sasl.username"]
+        password = config["sasl.password"]
+        method = SASLMethod[config["sasl.mechanism"].replace("-", "_")]
```

Comment:
We will likely only want to support `SHA-512` SCRAM auth. It might just be an error to provide anything else, soon. 

https://github.com/scimma/scimma-admin/blob/302b95d5f7d0468e0d4907231fd4f072b6d04352/scimma_admin/hopskotch_auth/models.py#L176

We could add that check after the scimma-admin auth mechanisms stabilize, though.

Reply:
Sounds good to me. This may be applicable to both comments but I can default to `SCRAM-SHA-512` if nothing is set.

Reply:
I ended up setting the default (`SCRAM-SHA-512`) in the previous comment but will tackle all the defaults/configuration stuff in a future PR.